### PR TITLE
Add username, groupname and external_id columns

### PR DESCRIPTION
**Pull request by tinagerber**
I did not migrate `username`/`groupname` for LDAP users/groups, because I think the mapping should be in the OGDS sync service. 

For local groups I set `groupid` as `groupname`, next OGDS sync should set `groupname`. As long as `groupname` can't be edited, that should be enough.

Also I am not 100% convinced that `username` and `groupname` should be unique, could probably lead to problems as inactive users/groups stay in the DB.

For [CA-4567]

## Checklist

_Everything has to be done/checked. Checked but not present means the author deemed it unnecessary._

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

_Only applicable should be left and checked._

- Upgrade-Steps:
  - [x] SQL Operations do not use imported model (see [docs](https://4teamwork.atlassian.net/wiki/spaces/4TEAM/pages/994344994/Upgrade-Steps))

[CA-4567]: https://4teamwork.atlassian.net/browse/CA-4567?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: changes/CA-4567-2.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+Add groupname and external_id columns to group model. [tinagerber]
```
Comment (lukasgraf) [inline changes/CA-4567-2.feature:1]: Double changelog?

File: opengever/ogds/models/user.py [hunk 2]
```diff
@@ -28,6 +28,9 @@ class User(Base):
     __tablename__ = 'users'
 
     userid = Column(String(USER_ID_LENGTH), primary_key=True)
+    username = Column(String(USER_ID_LENGTH), unique=True)
```
Comment (lukasgraf) [inline opengever/ogds/models/user.py:31]: I think you're right, `username` shouldn't have a unique constraint. After all, that's exactly why we're doing this - the username isn't a reliable primary key for users, and may be re-used for different accounts.

`external_id` should be unique however I think, at least at a given point in time.

File: opengever/ogds/models/user.py [hunk 3]
```diff
@@ -28,6 +28,9 @@ class User(Base):
...
+    external_id = Column(String(USER_ID_LENGTH), unique=True)
```
Comment (lukasgraf) [inline opengever/ogds/models/user.py:32]: Both `username` and `external_id` should be required (`nullable=False`). And as a consequence, everywhere where we create entries in the `users` or `groups` tables, we need to prefill `username` and `external_id` with whatever we write to `userid`. (Same for groups).

**Review by lukasgraf — CHANGES_REQUESTED** [review]
> I did not migrate username/groupname for LDAP users/groups, because I think the mapping should be in the OGDS sync service.

I don't quite understand what you mean by that.

File: changes/CA-4567-2.feature (hunk 1 above)
Comment (tinagerber) [reply changes/CA-4567-2.feature:1]: One is for the user model and the other for the group model 😄

**Comment by tinagerber** [comment]
@lukasgraf As discussed, I did not set an index for username and groupname because you are implementing a case insensitive index. The failing test also fails in the nightlies and has nothing to do with this implementation.
Can you have another look?

**Review by lukasgraf — APPROVED** [review]
LGTM 👍

**Review by buchi — APPROVED** [review]

**Pull request merged**